app/src/lib/stores/app-store.ts

```diff
@@ -664,6 +696,138 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.emitUpdate()
   }
 
+  private startAheadBehindUpdater(repository: Repository) {
+    if (this.currentAheadBehindUpdater) {
+      fatalError(
+        `An ahead/behind updater is already active and cannot start updating on ${
+          repository.name
+        }`
+      )
+
+      return
+    }
+
+    const updater = new AheadBehindUpdater(repository, aheadBehindCache => {
+      this.updateCompareState(repository, state => ({
+        aheadBehindCache,
+      }))
+      this.emitUpdate()
+    })
+
+    this.currentAheadBehindUpdater = updater
+
+    this.currentAheadBehindUpdater.start()
+  }
+
+  private stopAheadBehindUpdate() {
+    const updater = this.currentAheadBehindUpdater
+
+    if (updater) {
+      updater.stop()
+      this.currentAheadBehindUpdater = null
+    }
+  }
+
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _initializeCompare(
+    repository: Repository,
+    initialAction?: CompareAction
+  ) {
+    log.debug('[AppStore] initializing compare state')
+
+    const state = this.getRepositoryState(repository)
+
+    const branchesState = state.branchesState
+    const tip = branchesState.tip
+    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
+
+    const allBranches = currentBranch
+      ? branchesState.allBranches.filter(b => b.name !== currentBranch.name)
+      : branchesState.allBranches
+    const recentBranches = currentBranch
+      ? branchesState.recentBranches.filter(b => b.name !== currentBranch.name)
+      : branchesState.recentBranches
+
+    const cachedDefaultBranch = branchesState.defaultBranch
+
+    // only include the default branch when comparing if the user is not on the default branch
+    // and it also exists in the repository
+    const defaultBranch =
+      currentBranch != null &&
+      cachedDefaultBranch != null &&
+      currentBranch.name !== cachedDefaultBranch.name
+        ? cachedDefaultBranch
+        : null
+
+    this.updateCompareState(repository, state => ({
+      allBranches,
+      recentBranches,
+      defaultBranch,
+    }))
+
+    const compareState = state.compareState
+
+    const cachedState = compareState.formState
+
+    const action = initialAction ? initialAction : getInitialAction(cachedState)
+    this._executeCompare(repository, action)
+
+    if (currentBranch != null && this.currentAheadBehindUpdater != null) {
+      let allOtherBranches = [...recentBranches, ...allBranches]
```

Comment:
Yep, and we don't need to do any of this work because `allBranches` is already doing the  heavy lifting to exclude the current branch:

https://github.com/desktop/desktop/blob/34ece60342d5a0e6e3c7b07a4c1ef5f162fcc391/app/src/lib/stores/app-store.ts#L744-L747

Tidied up in 49debc372